src/pages/index.js

```diff
@@ -44,39 +39,92 @@ const GalleryItem = styled.div`
 `;
 
 const IndexPage = () => {
+  const [new_magazine, setNewMagazine] = useState([]);
+  const [thumbnail_image, setThumbnailImage] = useState([]);
+  useEffect(() => {
+    const sparkMagazine = async () => {
+      await fetch(
+        "https://api.github.com/repos/prince-deriv/deriv-static/contents/public/magazine"
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          if (result.length) {
+            const magazineDetails = [...result];
+            const data = magazineDetails.map(({ name, path }) => ({
+              name,
+              path,
+            }));
+
+            const months = [
+              "January",
+              "February",
+              "March",
+              "April",
+              "May",
+              "June",
+              "July",
+              "August",
+              "september",
+              "October",
+              "November",
+              "December",
+            ];
+
+            const filtered_data = data.filter((elem) => {
+              const [filtered_month, year] = elem.name.split("-");
+              if (
+                months.includes(filtered_month) &&
+                year.split(".")[0].length === 4
+              ) {
+                const month_year = filtered_month + `${"-"}` + year;
+                return elem.name === month_year;
+              }
+            });
+
+            setNewMagazine(filtered_data);
+          }
+        });
+    };
+    sparkMagazine();
+  }, []);
+  useEffect(() => {
+    const sparkMagazineImage = async () => {
+      await fetch(
+        "https://api.github.com/repos/prince-deriv/deriv-static/contents/public/thumbnails"
+      )
+        .then((res) => res.json())
+        .then((result) => {
+          if (result.length) {
+            const thumbnailImages = [...result];
+            setThumbnailImage(thumbnailImages);
+          }
+        });
+    };
+    sparkMagazineImage();
+  }, []);
```

Comment:
Let's do this in one `useEffect` also make sure that the thumbnail for specific month will reflect properly

ex: thumbnail/september-2022.png should be mapped to september-2022.pdf

Reply:
ok